ui/raidboss/data/06-ew/dungeon/the_tower_of_babil.txt

```diff
@@ -12,31 +12,75 @@ hideall "--sync--"
-72.2 "Dynamic Pound" sync / 1[56]:[^:]*:Barnabas:(6245|62EE):/ window 72.2,10
-73.3 "--sync--" sync / 1[56]:[^:]*:Barnabas:6075:/
-74.7 "Electromagnetic Release" sync / 1[56]:[^:]*:Barnabas:62EF:/
-85.3 "Thundercall" sync / 1[56]:[^:]*:Barnabas:62ED:/ window 85.3,10
-91.8 "Rolling Scrapline" sync / 1[56]:[^:]*:Barnabas:62EB:/
-92.1 "Shock" sync / 1[56]:[^:]*:Thunderball:62F2:/ window 92.1,10
-102.3 "Shocking Force" sync / 1[56]:[^:]*:Barnabas:62EC:/
-110.9 "Ground and Pound" sync / 1[56]:[^:]*:Barnabas:62EA:/
-116.9 "Ground and Pound" sync / 1[56]:[^:]*:Barnabas:62EA:/
-122.9 "Ground and Pound" sync / 1[56]:[^:]*:Barnabas:62EA:/
-
-138.8 "Dynamic Scrapline" sync / 1[56]:[^:]*:Barnabas:(6246|62F0):/ window 30,30 jump 40.5
-141.3 "Electromagnetic Release"
-153.5 "Shocking Force"
-170.5 "Dynamic Pound"
-173.0 "Electromagnetic Release"
-183.6 "Thundercall"
+# Barnabas can use either Dynamic Pound or Dynamic Scrapline first.
+# Unfortunately, the timing on the abilities up to Thundercall
+# is just offset enough to make it impractical to interleave the possibilities.
+# The Thundercall after the Pound/Scrapline blocks is the same
+# regardless of path taken.
+33.8 "--sync--" sync / 14:[^:]*:Barnabas:(6246|62F0):/ window 33.8,5 jump 193.3
+36.4 "--sync--" sync / 14:[^:]*:Barnabas:(6245|62EE):/ window 36.4,5 jump 293.3
+40.5 "Dynamic Scrapline?"
+43.0 "Electromagnetic Release?"
+43.1 "Dynamic Pound?"
+45.6 "Electromagnetic Release?"
+55.2 "Shocking Force"
+
+# Scrapline first
+193.3 "--sync--" sync / 14:[^:]*:Barnabas:(6246|62F0):/
+200.0 "Dynamic Scrapline" sync / 1[56]:[^:]*:Barnabas:(6246|62F0):/
+201.1 "--sync--" sync / 1[56]:[^:]*:Barnabas:61DD:/
+202.5 "Electromagnetic Release" sync / 1[56]:[^:]*:Barnabas:62F1:/
+214.7 "Shocking Force" sync / 1[56]:[^:]*:Barnabas:62EC:/ window 15,15
+231.7 "Dynamic Pound" sync / 1[56]:[^:]*:Barnabas:(6245|62EE):/
+232.8 "--sync--" sync / 1[56]:[^:]*:Barnabas:6075:/
+234.2 "Electromagnetic Release" sync / 1[56]:[^:]*:Barnabas:62EF:/
+
+244.8 "Thundercall" sync / 1[56]:[^:]*:Barnabas:62ED:/ window 44.8,5 jump 400
+251.3 "Rolling Scrapline"
+251.6 "Shock"
+261.8 "Shocking Force"
+270.4 "Ground and Pound 1"
+276.4 "Ground and Pound 2"
+
+# Pound first
+293.3 "--sync--" sync / 14:[^:]*:Barnabas:(6245|62EE):/
+300.0 "Dynamic Pound" sync / 1[56]:[^:]*:Barnabas:(6245|62EE):/
+301.1 "--sync--" sync / 1[56]:[^:]*:Barnabas:6075:/
+302.5 "Electromagnetic Release" sync / 1[56]:[^:]*:Barnabas:62EF:/
+314.7 "Shocking Force" sync / 1[56]:[^:]*:Barnabas:62EC:/ window 15,15
+329.9 "Dynamic Scrapline" sync / 1[56]:[^:]*:Barnabas:(6246|62F0):/
+331.0 "--sync--" sync / 1[56]:[^:]*:Barnabas:61DD:/
+332.4 "Electromagnetic Release" sync / 1[56]:[^:]*:Barnabas:62F1:/
+
+342.7 "Thundercall" sync / 1[56]:[^:]*:Barnabas:62ED:/ window 42.7,5 jump 400.0
+349.2 "Rolling Scrapline"
+349.5 "Shock"
+359.8 "Shocking Force"
+368.4 "Ground and Pound 1"
+374.4 "Ground and Pound 2"
+
+# Common Thhundercall block, looping back to one of the two blocks.
```

Comment:
```suggestion
# Common Thundercall block, looping back to one of the two blocks.
```

Does it always alternate after that, or is it totally random?

Reply:
It looks like it's a strict alternation of Scrapline --> Pound or Pound --> Scrapline *within the magnet block, but I think the timeline as it's written should handle any possibility?

Reply:
:+1: